ocpmodels/datasets/ase_datasets.py

```diff
@@ -0,0 +1,355 @@
+from pathlib import Path
+import ase
+import warnings
+import numpy as np
+from torch import tensor
+
+from torch.utils.data import Dataset
+from tqdm import tqdm
+
+from ocpmodels.common.registry import registry
+from ocpmodels.preprocessing import AtomsToGraphs
+
+
+def apply_one_tags(atoms, skip_if_nonzero=True, skip_always=False):
+    """
+    This function will apply tags of 1 to an ASE atoms object.
+    It is used as an atoms_transform in the datasets contained in this file.
+
+    args:
+        skip_if_nonzero (bool): If at least one atom has a nonzero tag, do not tag any atoms
+
+        skip_always (bool): Do not apply any tags. This arg exists so that this function can be disabled
+                without needing to pass a callable (which is currently difficult to do with main.py)
+    """
+    if skip_always:
+        return atoms
+
+    if np.all(atoms.get_tags() == 0) or not skip_if_nonzero:
+        atoms.set_tags(np.ones(len(atoms)))
+
+    return atoms
+
+
+class AseAtomsDataset(Dataset):
+    """
+    This is a base Dataset that includes helpful utilities for turning
+    ASE atoms objects into OCP-usable data objects.
+
+    Derived classes must add at least two things:
+        self.get_atoms_object(): a function that takes an identifier and returns a corresponding atoms object
+        self.id: a list of all possible identifiers that can be passed into self.get_atoms_object()
+    Identifiers need not be any particular type.
+    """
+
+    def __init__(self, config, transform=None, atoms_transform=apply_one_tags):
+        self.config = config
+
+        a2g_args = config.get("a2g_args", {})
+        self.a2g = AtomsToGraphs(
+            max_neigh=a2g_args.get("max_neigh", 1000),
+            radius=a2g_args.get("radius", 8),
```

Comment:
These defaults seem different from what we provide in `AtomsToGraphs`. Let's make them consistent here or there.

Reply:
While I don't necessarily agree with the defaults in AtomsToGraphs, I definitely agree with this design philosophy. However, leaving "r_edges" on does considerably slow down the speed of this dataset.

Changing AtomsToGraphs to not compute a graph by default does not seem like a good design choice. For this reason we should consider creating a more general "AtomsToData" class that builds on the generic targets work in #497. 

For now the default behavior is just to use the AtomsToGraphs default.

Reply:
I'm in favor of changing the `r_edges` default to False in AtomsToGraphs. Using `otf_graph=True` with our models is preferred since different models use different no. of neighbors / cutoffs etc. @mshuaibii @wood-b  thoughts?

Reply:
Other than `r_edges`, are you happy with the other defaults in AtomsToGraphs @emsunshine?

Reply:
In my opinion if `r_edges` defaults to False then `AtomsToGraphs` needs to be renamed. `r_pbc` is a tricky one because I would definitely want it here, but it would be wasteful to write the PBC for an OC20-sized LMDB. The cutoff and number of neighbors are also difficult because you'd want these to be at least as high as any model would actually use. Then again, if `r_edges` is False, they don't matter anyway.